core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1110,13 +1110,13 @@ object TestUtils extends Logging {
     values
   }
 
-  def produceMessage(servers: Seq[KafkaServer], topic: String, message: String) {
+  def produceMessage(servers: Seq[KafkaServer], topic: String, partition: Integer, message: String) {
```

Comment:
`partition` should be the last parameter and optional, maybe `partition: Option[Int] = None`?

Reply:
Currently constructor of ProducerRecord accepts partition as an Integer. And if the partition is null, then partition is considered as not specified. It seems intuitive and simple to just use the same pattern here.

I think either approach works for me and I don't have a strong opinion on this. If you like, I can submit a follow up patch to improve the code style as you suggested.

Reply:
`ProducerRecord` is in Java and Java doesn't have `Optional` until Java 8. In Scala, we avoid `nulls` because they're a major source of bugs.

Reply:
Also, `ProducerRecord` has overloads to simulate optional parameters.

Reply:
This parameter was needed by a previous version of this pull request. It seems that there is no need to have this parameter in the latest version. I will remove this parameter in pull request 3718.